### PR TITLE
Init project, keyboard-avoiding-view and pressable components

### PR DESCRIPTION
**Pull request by VitalyMih**
@ipakhomov I've set up this folder architecture for now. If we need a different one (for example, a custom plugin), I'll redo it.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    "react-native": "^0.74.1",
+    "react-native-safe-area-context": "^4.10.3",
+    "tslib": "^2.3.0"
+  },
+  "devDependencies": {
+    "@nx/eslint": "19.1.1",
+    "@nx/eslint-plugin": "19.1.1",
+    "@nx/js": "19.1.1",
+    "@nx/vite": "19.1.1",
+    "@nx/web": "19.1.1",
```
Comment (ipakhomov) [inline package.json:21]: Not sure we need these dependencies. What the template have you used?
I think we could use the [NX Expo template](https://nx.dev/nx-api/expo) and have:
- some demo app to see and test the components in action. This app won't be published to NPM
- the `src/lib` with common modules will be a [publishable library](https://nx.dev/concepts/buildable-and-publishable-libraries#publishable-libraries) to distribute in over NPM

File: src/lib/ui/keyboard-avoiding-view/component.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { KeyboardAwareScrollView } from '@pietile-native-kit/keyboard-aware-scrollview';
+import { ReactElement, ReactNode } from 'react';
+import { KeyboardAvoidingView, Platform, ScrollViewProps, StyleSheet, ViewStyle } from 'react-native';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+
+export interface AppKeyboardAvoidingViewProps extends Partial<ScrollViewProps> {
+  children: ReactNode;
+  withoutScrollView?: boolean;
```
Comment (ipakhomov) [inline src/lib/ui/keyboard-avoiding-view/component.tsx:8]: How about? To match `ScrollViewProps`:

```suggestion
  scrollEnabled?: boolean;
```

File: src/lib/ui/keyboard-avoiding-view/component.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { KeyboardAwareScrollView } from '@pietile-native-kit/keyboard-aware-scrollview';
+import { ReactElement, ReactNode } from 'react';
+import { KeyboardAvoidingView, Platform, ScrollViewProps, StyleSheet, ViewStyle } from 'react-native';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
+
+export interface AppKeyboardAvoidingViewProps extends Partial<ScrollViewProps> {
```
Comment (ipakhomov) [inline src/lib/ui/keyboard-avoiding-view/component.tsx:6]: If `withoutScrollView` will be true, this component won't extend the `ScrollViewProps`. Could you please make an overload for props?

Example:

```tsx
export interface AppKeyboardAvoidingViewCommonProps {
  scrollEnabled?: boolean;
  extraHeight?: number;
  contentContainerStyle?: ViewStyle;
}

interface AppKeyboardAvoidingScrollableViewProps extends ScrollViewProps {
  scrollEnabled: true;
}

interface AppKeyboardAvoidingNonScrollableViewProps extends ViewProps {
  scrollEnabled?: boolean;
}

export type AppKeyboardAvoidingViewProps = AppKeyboardAvoidingViewCommonProps &
  (AppKeyboardAvoidingScrollableViewProps | AppKeyboardAvoidingNonScrollableViewProps);
```

File: src/lib/ui/pressable/component.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { Pressable, PressableProps, StyleSheet, StyleProp, ViewStyle } from 'react-native';
+
+export interface AppPressableProps extends PressableProps {
+  style?: StyleProp<ViewStyle>;
+}
+
+export function AppPressable({ children, style, ...props }: AppPressableProps) {
+  return (
+    <Pressable style={({ pressed }) => StyleSheet.flatten([{ opacity: pressed ? 0.4 : 1 }, style])} {...props}>
```
Comment (ipakhomov) [inline src/lib/ui/pressable/component.tsx:10]: To make component more flexible, we could allow customizing `pressedOpacity`:

```suggestion
    <Pressable style={({ pressed }) => StyleSheet.flatten([{ opacity: pressed ? pressedOpacity : 1 }, style])} {...props}>
```

File: src/lib/ui/pressable/component.tsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { Pressable, PressableProps, StyleSheet, StyleProp, ViewStyle } from 'react-native';
+
+export interface AppPressableProps extends PressableProps {
+  style?: StyleProp<ViewStyle>;
```
Comment (ipakhomov) [inline src/lib/ui/pressable/component.tsx:5]: Isn't `style` props presented in `PressableProps`? What's the purpose of `AppPressableProps`?

**Review by ipakhomov — CHANGES_REQUESTED** [review]
@VitalyMih thank you for your contribution! Please find some suggestions from me

File: .prettierrc [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+  "singleQuote": true
```
[the remark this answers is not in the document]
Comment (VitalyMih) [reply .prettierrc]: Done

File: package.json (hunk 1 above)
Comment (VitalyMih) [reply package.json]: Done

File: src/lib/ui/keyboard-avoiding-view/component.tsx (hunk 2 above)
Comment (VitalyMih) [reply src/lib/ui/keyboard-avoiding-view/component.tsx]: As discussed - removed

File: src/lib/ui/keyboard-avoiding-view/component.tsx (hunk 3 above)
Comment (VitalyMih) [reply src/lib/ui/keyboard-avoiding-view/component.tsx]: As discussed - removed

**Comment by VitalyMih** [comment]
@ipakhomov check again, please :)

**Review by ipakhomov — CHANGES_REQUESTED** [review]

File: src/lib/package.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "@react-native-common-modules",
+  "version": "0.0.1",
+  "main": "src/index.ts",
```
Comment (ipakhomov) [inline src/lib/package.json:4]: I think we should also specify which files are included into the package using `files` array - [example](https://github.com/RonasIT/rtkq-entity-api/blob/main/package.json#L28C3-L30C5)

File: src/lib/package.json [hunk 8]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "@react-native-common-modules",
```
Comment (ipakhomov) [inline src/lib/package.json:2]:
```suggestion
  "name": "@ronas-it/react-native-common-modules",
```

**Review by ipakhomov — CHANGES_REQUESTED** [review]

**Comment by ipakhomov** [comment]
@VitalyMih few more comments

File: .vscode/extensions.json
[the remark this answers is not in the document]
Comment (VitalyMih) [reply .vscode/extensions.json:1]: Sorry, fixed

**Comment by VitalyMih** [comment]
@ipakhomov fixed :) check again, please

**Pull request merged**